v-next/hardhat-keystore/src/internal/ui/user-interactions.ts

```diff
@@ -0,0 +1,120 @@
+import type { UserInterruptionManager } from "@ignored/hardhat-vnext/types/user-interruptions";
+
+import chalk from "chalk";
+
+import { PLUGIN_ID } from "../constants.js";
+
+export class UserInteractions {
+  readonly #userInterruptions: UserInterruptionManager;
+
+  constructor(userInterruptions: UserInterruptionManager) {
+    this.#userInterruptions = userInterruptions;
+  }
+
+  public async setUpPassword(): Promise<string> {
+    const PASSWORD_REGEX = /^(?=.*[A-Z])(?=.*[a-z])(?=.*[\W_]).{8,}$/;
+
+    const setupMsg =
+      "This is the first time you are using the keystore, please set a password.";
+    const passwordRulesMsg =
+      "The password must have at least 8 characters, one uppercase letter, one lowercase letter, and one special character.";
+
+    await this.#displayMessage("\n👷🔐 Hardhat-Keystore 🔐👷\n");
+    await this.#displayMessage(setupMsg);
+    await this.#displayMessage(passwordRulesMsg);
+    await this.#displayMessage("");
+
+    let password: string | undefined;
+
+    while (password === undefined) {
```

Comment:
What's this while here? What does `undefined` mean?

Reply:
Small note: this code is currently not used, it will be used when the encryption is enabled.

To answer to your question:

There are two steps when setting up the password for the keystore:
1. the user must enter a password
2. the user must confirm the password

in both scenarios there is a while lookup, why?
- step 1: loop until the password is valid -it must match the regex (the process can be interrupted by the user)
- step 2: loop until the passwords do NOT match (the process can be interrupted by the user)

Reply:
To be revisited when we implement encryption